### PR TITLE
Add `updateTexture` & some shortcuts in Sprite; Cosmetic

### PR DESCRIPTION
**Pull request by finscn**
* Add `updateText`.  If a sprite is created by a canvas ,  when the canvas's changed , user could call `updateText` for updating sprite.
* Add some shortcuts for accessing  some properties of sprite._texture.
* Cosmetic Text. Just define some local vars for accessing properties of `this` , for DRY.

### CODE REVIEW

**Comment by ivanpopelyshev** [comment]
Agree on "this" in Text, but updateTexture.. hm. Its handy for single sprites that are created from canvases, but I dont know if its good to put it there.

@GoodBoyDigital @bigtimebuddy what about "SpriteCanvas" class, that is created from canvas and not from texture? Or with canvas. Move part of Text there, add this updateTexture method.

**Comment by finscn** [comment]
How about these `shortcuts` ?

**Comment by finscn** [comment]
And If use SpriteCanvas ,  do we need  TextureCanvas & BaseTextureCanvas too ???

**Comment by ivanpopelyshev** [comment]
@finscn definitely no. Only for sprite.

**Comment by bigtimebuddy** [comment]
I like the local variable caching, that should help minifying output. Small, but good change.

I disagree with adding `baseTexture`, `orig`, `frame`, and `trim` getters for Sprite. It's bloats the API and doesn't provide a lot of convenience. These things are already publicly available `sprite.texture.baseTexture`, `sprite.texture.orig`, etc.

Also, I don't understand the need to add `updateTexture`. This functionality is also already available: 
`sprite.texture.baseTexture.update()` (example: https://jsfiddle.net/bedrdc1b/).

So in summary, I like the local variable caching, but disagree with the API changes.

**Comment by bigtimebuddy** [comment]
ping @ivanpopelyshev
